bento/src/main/java/com/yelp/android/bento/components/ListComponent.java

```diff
@@ -86,17 +86,31 @@ public void appendData(@NonNull List<T> data) {
     }
 
     /**
-     * Removes the provided data items from the list.
+     * Inserts a single list item at the specified index in the list.
+     *
+     * @param index The index used to insert the given data in the list.
+     * @param data The list item to be inserted.
+     */
+    public void insertData(int index, @NonNull T data) {
+        mData.add(index, data);
+        // Update 2 items if dividers are showing.
+        notifyItemRangeInserted(index, mShouldShowDivider ? 2 : 1);
```

Comment:
if divider are visible, you probably need to multiply index by 2, otherwise you will have some surprises